src/app/widgets/Visualizer/index.jsx

```diff
@@ -1128,6 +1128,7 @@ class VisualizerWidget extends PureComponent {
                     showVisualizer={showVisualizer}
                     visualizerRef={(ref) => {
                         this.visualizer = ref;
+                        console.log(ref);
```

Comment:
Instead of setting **this.visualizer** to the exact ref of the visualizer, you can just set it to the visualizer property of the ref, since it is present as a ref in the visualizer wrapper component you set up and is being passed to the primary/secondary visualizer components. ( ex. **this.visualizer = ref?.visualizer**, we add optional chaining here so we don't run into problems when the ref has not been set yet and doesn't have a visualizer property )